org.lflang/src/org/lflang/generator/cpp/CppConnectionGenerator.kt

```diff
@@ -42,18 +69,24 @@ class CppConnectionGenerator(private val reactor: Reactor) {
         reactor.connections.mapNotNull { generateConstructorInitializer(it) }.joinLn()
 
     private fun generateDecleration(connection: Connection): String? =
-        if (connection.requiresConnectionClass) {
-            if (connection.hasMultipleConnections) {
-                "std::vector<${connection.cppType}> ${connection.name};"
-            } else {
-                "${connection.cppType} ${connection.name};"
+        with(connection) {
+            if (requiresConnectionClass) {
+                when {
+                    hasMultipleConnections && isEnclaveConnection  -> "std::list<${connection.cppType}> ${connection.name};"
+                    hasMultipleConnections && !isEnclaveConnection -> "std::vector<${connection.cppType}> ${connection.name};"
```

Comment:
why do you generate here different containers ?

Reply:
I don't remember exactly, but I think the problem was that vector requires that the elements are movable. I will check this again and should add a comment here

Reply:
The problem was indeed that connections are not default constructable and not safely movable at the moment. However, I fall back on using a vector of unique pointers to connections now.